src/sc/sidechaintypes.cpp

```diff
@@ -0,0 +1,157 @@
+                std::to_string(scFieldElement.size()) + "]"));
+        }
+        // pad with zeroes, must have the same size for the internal repr
+        std::vector<unsigned char> temp(vRawField);
+        temp.resize(scFieldElement.size(), 0x0);
+        *const_cast<libzendoomc::ScFieldElement*>(&scFieldElement) = libzendoomc::ScFieldElement(temp); // TODO
+    }
+}
+
+const libzendoomc::ScFieldElement& FieldElement::GetFieldElement() const
+{
+    InitFieldElement();
+    return scFieldElement;
+}
+
+#ifdef BITCOIN_TX
+bool FieldElement::IsValid() const { return true; }
+#else
+bool FieldElement::IsValid() const
+{
+    InitFieldElement();
+    if (scFieldElement.IsNull())
+        return false;
+
+    return libzendoomc::IsValidScFieldElement(scFieldElement);
+};
+#endif
+
+bool FieldElement::checkCfg(const CustomFieldConfig& cfg) const
+{
+    if (vRawField.size() != cfg.getBitSize() )
+    {
+        LogPrint("sc", "%s():%d - ERROR: wrong size: data[%d] != cfg[%d]\n", 
+            __func__, __LINE__, vRawField.size(), cfg.getBitSize());
+        return false;
+    }
+    return true;
+};
+
+//----------------------------------------------------------------------------------
+CompressedMerkleTree::CompressedMerkleTree(const CompressedMerkleTreeConfig& cfg)
+    :CustomField(cfg) {}
+
+CompressedMerkleTree::CompressedMerkleTree(const std::vector<unsigned char>& rawBytes)
+    :CustomField(rawBytes) {}
+
+CompressedMerkleTree& CompressedMerkleTree::operator=(const CompressedMerkleTree& rhs)
+{
+    *const_cast<std::vector<unsigned char>*>(&vRawField) = rhs.vRawField;
+    return *this;
+}
+
+void CompressedMerkleTree::InitFieldElement() const
+{
+    if (merkleRoot.IsNull())
+    {
+        if (vRawField.size() > merkleRoot.size())
+        {
+            LogPrint("sc", "%s():%d - ERROR: wrong size: rawData[%d] > mklRoot[%d]\n", 
+                __func__, __LINE__, vRawField.size(), merkleRoot.size());
+            throw std::invalid_argument(
+                std::string("Wrong size: rawData[" + std::to_string(vRawField.size()) + "] > mklRoot[" +
+                std::to_string(merkleRoot.size()) + "]"));
+        }
+        // pad with zeroes, must have the same size for the internal fe repr
+        std::vector<unsigned char> temp(vRawField);
+        temp.resize(merkleRoot.size(), 0x0);
+        *const_cast<libzendoomc::ScFieldElement*>(&merkleRoot) = libzendoomc::ScFieldElement(temp); // TODO
+    }
+}
+
+const libzendoomc::ScFieldElement& CompressedMerkleTree::GetFieldElement() const
+{
+    InitFieldElement();
+    return merkleRoot;
+}
+
+bool CompressedMerkleTree::IsValid() const
+{
+    InitFieldElement();
+    if (merkleRoot.IsNull())
+        return false;
+
+    // TODO something like libzendoomc::IsValidScFieldElement() or exactly this?? In this case we can move this to base   
+    return true;
+}
+
+bool CompressedMerkleTree::checkCfg(const CustomFieldConfig& cfg) const
+{
+    if (vRawField.size() > cfg.getBitSize() )
```

Comment:
same as for `FieldElement` class bytes vs bits

Reply:
Done